HPXMLtoOpenStudio/resources/electric_panel.rb

```diff
@@ -0,0 +1,237 @@
+      next if panel_load.type != HPXML::ElectricPanelLoadTypeHeating
+
+      heating_system = get_panel_load_heating_system(hpxml_bldg, panel_load)
+      if !heating_system.nil?
+        heating_system_watts = panel_load.power
+        primary_heat_pump_watts = 0
+        if !heating_system.primary_heat_pump.nil?
+          primary_heat_pump_watts = electric_panel.panel_loads.find { |pl| pl.system_idrefs.include?(heating_system.primary_heat_pump.id) }.power
+        end
+
+        if addition.nil? ||
+           (addition && panel_load.addition) ||
+           (!addition && !panel_load.addition)
+          if (primary_heat_pump_watts == 0) ||
+             (!heating_system.primary_heat_pump.nil? && heating_system.primary_heat_pump.simultaneous_backup) ||
+             (!heating_system.primary_heat_pump.nil? && heating_system_watts >= primary_heat_pump_watts)
+            htg += heating_system_watts
+          end
+        end
+      end
+
+      heat_pump = get_panel_load_heat_pump(hpxml_bldg, panel_load)
+      next unless !heat_pump.nil?
+
+      heat_pump_watts = panel_load.power
+      backup_system_watts = 0
+      if !heat_pump.backup_system.nil?
+        backup_system_watts = electric_panel.panel_loads.find { |pl| pl.system_idrefs.include?(heat_pump.backup_system.id) }.power
+      end
+
+      next unless addition.nil? ||
+                  (addition && panel_load.addition) ||
+                  (!addition && !panel_load.addition)
+
+      next unless (backup_system_watts == 0) ||
+                  (!heat_pump.backup_system.nil? && heat_pump.simultaneous_backup) ||
+                  (!heat_pump.backup_system.nil? && heat_pump_watts >= backup_system_watts)
+
+      htg += heat_pump_watts
+    end
+    return htg
+  end
+
+  # Calculate the load-based capacity for the given electric panel and panel loads according to NEC 220.83.
+  #
+  # @param hpxml_bldg [HPXML::Building] HPXML Building object representing an individual dwelling unit
+  # @param electric_panel [HPXML::ElectricPanel] Object that defines a single electric panel
+  # @param panel_loads [Array<HPXML::PanelLoad>] List of panel load objects
+  # @return [nil]
+  def self.calculate_load_based(hpxml_bldg, electric_panel, panel_loads, panel_calculation_type)
+    if panel_calculation_type == HPXML::ElectricPanelLoadCalculationType2023LoadBased
+      htg_existing = get_panel_load_heating(hpxml_bldg, electric_panel, addition: false)
+      htg_new = get_panel_load_heating(hpxml_bldg, electric_panel, addition: true)
+      clg_existing = electric_panel.panel_loads.select { |panel_load| panel_load.type == HPXML::ElectricPanelLoadTypeCooling && !panel_load.addition }.map { |pl| pl.power }.sum(0.0)
+      clg_new = electric_panel.panel_loads.select { |panel_load| panel_load.type == HPXML::ElectricPanelLoadTypeCooling && panel_load.addition }.map { |pl| pl.power }.sum(0.0)
+
+      # Part A
+      other_load = [htg_existing, clg_existing].max
```

Comment:
Additionally, we need to special handle 120V dryer.

Currently we assign one 1.5kVA (20A) circuit for laundry, which will support the washer (requires 12A on avg). We should check that if the laundry circuit can accommodate the 120V dryer, we remove that load from panel_loads. I think this will almost always going to be no, but we should add that adjustment.

Would we do the adjustment here?

Reply:
BTW, where is the kitchen and laundry circuit VA specified in the code?

Reply:
@lixiliu  - kitchen and laundry VA are specified in the [`defaults.rb`](https://github.com/NREL/OpenStudio-HPXML/blob/3cf18f50162ae6e5be6a575371e9321613433a4c/HPXMLtoOpenStudio/resources/defaults.rb#L3350-L3358) file